activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1138,6 +1168,25 @@ def preprocess_order_args(order_args)
         end.flatten!
       end
 
+      # Only allow column names and directions as arguments to #order and
+      # #reorder. Other arguments will cause an ArugmentError to be raised.
+      def restrict_order_args(args)
+        args = args.dup
+        orderings = args.extract_options!
+        columns = args | orderings.keys
+
+        unrecognized = columns.reject { |c| klass.respond_to_attribute?(c) }
+        if unrecognized.any?
+          raise ArgumentError, "Invalid order column: #{unrecognized}"
+        end
+
+        # TODO: find a better list of modifiers.
+        unrecognized = orderings.values.reject { |d| VALID_DIRECTIONS.include?(d.to_s) }
+        if unrecognized.any?
+          raise ArgumentError, "Invalid order direction: #{unrecognized}"
```

Comment:
Consider `unrecognized.inspect` so the message reflects the string/symbol syntax from the caller.